pkg/reconciler/v1alpha1/route/resources/virtual_service.go

```diff
@@ -150,6 +151,10 @@ func makeVirtualServiceRoute(domains []string, ns string, targets []traffic.Revi
 		Match:   matches,
 		Route:   weights,
 		Timeout: DefaultRouteTimeout,
+		Retries: &v1alpha3.HTTPRetry{
+			Attempts:      DefaultRouteRetryAttempts,
+			PerTryTimeout: DefaultRouteTimeout,
```

Comment:
I am OK with adding a retry since this could mitigate some problems we got. 

But I am not sure what is the actual behavior for below case.
Assume an http call from Service A to service B always takes 40 seconds, and then fails with 404 error. If we add retry, will the client (Service A) get a real error code (404) or timeout error (since the second request takes another 40 seconds, and exceeds the timeout seconds)? If the client gets the timeout error, which seems to be confusing.

Reply:
Yeah, I think that will happen.  However, that will be due to slow requests (3 attempts taking 60 seconds) so timeout (504) is partially true.

I believe we don't retry on 4xx, only 5xx which are usually returning fast (503: upstream reset, upstream connect error, no healthy upstream...) unless they are 504s in which case we are returning the correct thing.

https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/route/route.go#L521